ShipMonitor/ModuleInfoReader.cs

```diff
@@ -0,0 +1,118 @@
+﻿using EddiDataDefinitions;
+using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Text.RegularExpressions;
+using Utilities;
+
+
+
+namespace EddiShipMonitor
+{
+    public class ModuleInfoReader
+    {
+        public List<ModuleInfo> Modules { get; set; }
+
+        [JsonIgnore]
+        private string dataPath;
+
+        [JsonIgnore]
+        static readonly object fileLock = new object();
+
+        public ModuleInfoReader()
+        {
+            Modules = new List<ModuleInfo>();
+        }
+
+        public static ModuleInfoReader FromFile(string filename = null)
+        {
+        Regex Filter = new Regex(@"^ModulesInfo\.json$");
+        string directory = GetSavedGamesDir();
+            if (directory == null || directory.Trim() == "")
```

Comment:
For future reference, there is `String.IsNullOrWhiteSpace()` to for convenience.